crates/sparrow-compiler/src/ast_to_dfg.rs

```diff
@@ -691,6 +697,17 @@ fn cast_if_needed(
     }
 }
 
+fn map_types_are_equal(a: &FieldRef, b: &FieldRef) -> bool {
```

Comment:
This was to get around the issue where maps
```
map(name: "m1", field: ..)
map(name: "m2", field: ..)
```
were not equal because the `name` was different. However, since we infer the concrete datatype in `type_inference`, we create a concrete `DataType::Map`, meaning we have to assign default namings. This was causing the inequality in map types that should be equal. 

I wonder if there's some edge case I'm not thinking of where this matters, though. I don't think so, but have done little exploration of this.

Reply:
Add some more comments to that effect so we can reconstruct why we're doing this?

Reply:
Can you put more of that comment on the method -- explain what the method does / why it exists (and then you just need a short comment on where it is used to explain why we need it there)?

Reply:
Added